docs/DeveloperGuide.md

```diff
@@ -479,8 +485,68 @@ Given below are instructions to test the app manually.
+    Details of the success of the edit task will be shown.
+    2. Test case: `edit -t 10 meet girlfriend`, where there is no task `10` in the task list.\
+    Expected: As there is no such task in the task list, an error will be given to the user. Details of the associated error 
+    message will be shown.
+
+### Editing a module
+1. Editing a module 
+    1. Test case: `edit -m CG2271 GER1000`, where module `CG2271` has already been added previously into the module list.\
+    Expected: As `CG2271` is in the module list, the module will be edited and changed to `GER1000`. Details of the success 
+    of the edit module will be shown.
+    2. Test case: `edit -m ACC1101 GER1000`, where there is no module `ACC1101` in the module list.\
+    Expected: As there is no such module in the module list, an error will be given to the user. Details of the associated 
+    error message will be shown. 
+    
+### Marking a task as done / undone
+1. Marking a task as done
+    1. Test case: `done 1`, where `task` is index `0` in the task list.\
+    Expected: The DoneCommandParser parses `1` and converts it to index `0` in the task list. 
+    As task `task` is the index `0` in the task list, `task` will be marked as done.
+    2. Test case: `done 10`, where there is no task of index `9` in the task list.\
+    Expected: The DoneCommandParser parses `10` and converts it to index `9` in the task list. 
+    As there is no task of index `9` in the task list, an error is thrown. Details of the associated error message will be shown.
+  
+### Viewing the task list
+1. Viewing the task list
+    1. Test case: `list -t`, after adding the tasks "read book", "buy stuff" have been added in succession.\
+    Expected: A list containing the two tasks will be shown.
+    2. Test case: `list -t` when no tasks have been added.\
+    Expected: As there are no tasks in the list, a message signifying the empty list is shown.
+    
+### Viewing the module list
+1. Viewing the module list
+    1. Test case: `list -m`, after adding the modules "CS2113T", and "CS2101" have been added in succession.\
+    Expected: A list containing the two module codes, MCs, and grades associated will be shown.
+    2. Test case: `list -m` when no modules have been added.\
+    Expected: As there are no modules in the list, a message signifying the empty list is shown. 
+
+### Viewing help messages and prompts
+1. Viewing the help message for a specific command
+    1. Test case: `help add`\
+    Expected: A help message containing the `add` command format and examples of its usage is shown.
+    2. Test case: `add CS2101`\
+    Expected: As this is an incorrect command format, a help message containing the correct format is shown. A prompt message suggesting the use of `help` is also shown. 
     
-### Amend the timetable
+### Viewing task summary
+1. Viewing the task summary 
+    1. Test case: 
+    `summary` after inputting the following:
```

Comment:
Rather than input, maybe say these tasks are in the task list, e.g. show what is in the task list